dvc_ssh/__init__.py

```diff
@@ -51,44 +54,25 @@ def _prepare_credentials(self, **config):
         login_info["client_factory"] = config.get(
             "client_factory", InteractiveSSHClient
         )
-        try:
-            user_ssh_config = parse_config(host=config["host"])
-        except FileNotFoundError:
-            user_ssh_config = {}
-
-        login_info["host"] = user_ssh_config.get("Hostname", config["host"])
-
-        login_info["username"] = (
-            config.get("user")
-            or config.get("username")
-            or user_ssh_config.get("User")
-            or getpass.getuser()
-        )
-        login_info["port"] = (
-            config.get("port")
-            or silent(int)(user_ssh_config.get("Port"))
-            or DEFAULT_PORT
-        )
+        login_info["host"] = config["host"]
+        if username := (config.get("user") or config.get("username")):
+            login_info["username"] = username
+        if port := config.get("port"):
+            login_info["port"] = port
 
         for option in ("password", "passphrase"):
             login_info[option] = config.get(option)
 
             if config.get(f"ask_{option}") and login_info[option] is None:
                 login_info[option] = ask_password(
                     login_info["host"],
-                    login_info["username"],
-                    login_info["port"],
+                    login_info.get("username"),
+                    login_info.get("port"),
                     option,
                 )
 
-        raw_keys = []
-        if config.get("keyfile"):
-            raw_keys.append(config.get("keyfile"))
-        elif user_ssh_config.get("IdentityFile"):
-            raw_keys.extend(user_ssh_config.get("IdentityFile"))
-
-        if raw_keys:
-            login_info["client_keys"] = [os.path.expanduser(key) for key in raw_keys]
+        if keyfile := config.get("keyfile"):
+            login_info["client_keys"] = [os.path.expanduser(keyfile)]
```

Comment:
One thing worth mentioning, if `keyfile` is set and `passphrase`/`ask_passphrase` is not set, DVC fails without asking for a passphrase.  This is working as expected, but we do prompt for passphrase for default keyfiles or the one specified in the user's config. So, we are diverging a bit.

`asyncssh` can take a `passphrase` callable, so if we pass it, it will also get called for these `client_keys`. This will unify the behavior and remove the need for eager `ask_passphrase` callback above.

Will create a PR for this separately.

Reply:
nvmd, ignore that. `passphrase` callable will be called for every key, even unauthenticated ones.